### PR TITLE
Improve scheduling, persist node details and add some tests

### PR DESCRIPTION
**Pull request by w-miller**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,22 @@ installed inside it, Tenks' role dependencies can be installed by
...
+Tenks uses Ansible inventory to manage hosts. A multi-host setup is therefore
+supported, although the default hosts configuration will deploy an all-in-one
+setup on the host where the `ansible-playbook` command is executed
+(`localhost`).
+
+* Configuration management of the Tenks cluster is always performed on
+  `localhost`.
+* The `hypervisors` group should not directly contain any hosts. Its sub-groups
+  must contain one or more system. Systems in its sub-groups will host a subset
+  of the noides deployed by Tenks.
```
Comment (markgoddard) [inline README.md]: nodes

File: tests/test_tenks_update_state.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        for node in expected_state['foo']['nodes']:
+            node['state'] = 'absent'
+        self.mod.localhost_vars['cmd'] = 'teardown'
+        self.mod._process_specs()
+        self.assertEqual(expected_state, self.args['state'])
+
+        # After yet another run, the 'absent' state nodes should be deleted
+        # from state altogether.
+        self.mod._process_specs()
+        self.assertEqual(self.args['state']['foo']['nodes'], [])
```
Comment (markgoddard) [inline tests/test_tenks_update_state.py]: Does this imply that scheduling twice in a row will remove the node from the state? What if the playbook fails?

File: README.md (hunk 1 above)
Comment (w-miller) [reply README.md]: Thanks.

File: tests/test_tenks_update_state.py (hunk 2 above)
Comment (w-miller) [reply tests/test_tenks_update_state.py]: Yes, unfortunately it does. I couldn't really think of a better way to do this, have you got any ideas?

Comment (w-miller) [reply tests/test_tenks_update_state.py]: I suppose we could leave absent nodes in the inventory forever but it might get a bit crufty...

Comment (markgoddard) [reply tests/test_tenks_update_state.py]: You could perform a second scheduling action at the end of the deploy.yml playbook that removes the nodes. Could modify the action to only remove absent nodes if a (new) flag is set.

Comment (w-miller) [reply tests/test_tenks_update_state.py]: That's true - I suppose that wouldn't even need to go through the Python action plugin if we just read from the file, used a Jinja `select` to remove absent nodes and wrote back to the file.

**Comment by w-miller** [comment]
Travis checks finally working again - updated and ready for re-review.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**